runtime/builtin_types.go

```diff
@@ -580,6 +580,14 @@ func builtinSorted(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
 		return nil, raised
 	}
 	toListUnsafe(result).Sort(f)
+	// Implement reverse. 
+	reverse, raised := IsTrue(f, kwargs.get("reverse", None))
+	if raised != nil {
+		return nil, raised
+	}
+	if reverse {
+		toListUnsafe(result).Reverse()
```

Comment:
As you mention, stability of the results is a concern here. If we supported the cmp param, we could negate the result of that to achieve reversal. This would mean that listSorter would need to support cmp. Then sorted(... reverse=True) could wrap that cmp and negate the result (or provide a stock reverse cmp when cmp=None).

So maybe it makes sense to implement cmp first?